app/models/case_information.rb

```diff
@@ -23,22 +25,67 @@ def latest_early_allocation
   validates :manual_entry, inclusion: { in: [true, false], allow_nil: false }
   validates :nomis_offender_id, presence: true, uniqueness: true
 
-  validates :local_divisional_unit, :team, presence: true, unless: ->{ manual_entry }
+  validates :team,
+            presence: { message: "You must select the prisoner's team" },
+            unless:
+            proc { |c|
+              c.manual_entry &&
+              (c.probation_service == 'Scotland' ||
+              c.probation_service == 'Northern Ireland')
+            }
 
-  validates :welsh_offender, inclusion: {
-    in: %w[Yes No],
-    allow_nil: false,
-    message: 'Select yes if the prisoner’s last known address was in Wales'
-  }
-  validates :tier, inclusion: { in: %w[A B C D], message: 'Select the prisoner’s tier' }
+  validates :tier, inclusion: { in: %w[A B C D N/A], message: "Select the prisoner's tier" }
 
   validates :case_allocation, inclusion: {
-    in: %w[NPS CRC],
+    in: %w[NPS CRC N/A],
     allow_nil: false,
     message: 'Select the service provider for this case'
   }
 
   # nil means MAPPA level is completely unknown.
   # 0 means MAPPA level is known to be not relevant for offender
   validates :mappa_level, inclusion: { in: [0, 1, 2, 3], allow_nil: true }
+
+  validates :probation_service, inclusion: {
+    in: ['Scotland', 'Northern Ireland', 'Wales', 'England'],
+    allow_nil: false,
+    message: "You must say if the prisoner's last known address was in Northern Ireland, Scotland or Wales"
+  }, if: -> { manual_entry }
+
+  # We only want to validate last known location in forms
+  validates :last_known_location,
+            inclusion: {
+              in: %w[Yes No],
+              allow_nil: true,
+              message: "Select yes if the prisoner's last known address was in Northern Ireland, Scotland or Wales"
+            }, if: -> { manual_entry }
+
+  def scottish_or_ni?
+    return true if last_known_location == 'Yes' &&
+      (probation_service == 'Scotland' || probation_service == 'Northern Ireland')
+
+    false
+  end
+
+  def save_scottish_or_ni
+    self.tier = 'N/A'
+    self.case_allocation = 'N/A'
+    self.team = nil
+  end
+
+  def english_or_welsh?
+    return true if last_known_location == 'No' || probation_service == 'Wales' || probation_service == 'England'
+
+    false
+  end
+
+  def english?
+    last_known_location == 'No'
+  end
+
+  def stage2_filled?
+    return false if tier.nil? || team.nil? || case_allocation.nil?
+
+    true
+  end
```

Comment:
I think I've answered my own question here. Are `nil` values in the 'tier' and 'case allocation' fields reserved to mean "The user hasn't completed this field", rather than "The user chose N/A"?

Also – I wonder if this could be easier to read if the logic is flipped:
```suggestion
  def stage2_filled?
    tier.present? && team.present? && case_allocation.present?
  end
```